src/macaron/dependency_analyzer/dependency_resolver.py

```diff
@@ -173,16 +178,29 @@ def add_latest_version(
                 logger.error("Could not parse dependency version number: %s", error)
 
     @staticmethod
-    def _find_repo(item: DependencyInfo) -> None:
+    def _find_repo(db_man: DatabaseManager | None, item: DependencyInfo) -> None:
         """Find the repo for the current item, if the criteria are met."""
         if item["url"] != "" or item["version"] == "unspecified" or not item["group"] or not item["name"]:
             logger.debug("Item URL already exists, or item is missing information: %s", item)
             return
-        gav = f"{item['group']}:{item['name']}:{item['version']}"
+        artifact = f"{item['group']}:{item['name']}"
         if f"{item['group']}:{item['name']}" in defaults.get_list("repofinder.java", "artifact_ignore_list"):
-            logger.debug("Skipping GAV: %s", gav)
+            logger.debug("Skipping artifact: %s", artifact)
             return
 
+        if db_man and defaults.getboolean("repofinder.java", "use_database"):
+            # Perform database lookup
+            query = sqlalchemy.text(
+                "SELECT remote_path FROM _repository WHERE namespace = :group and name = :artifact"
```

Comment:
Instead of raw queries, we should use SQLAlchemy's [ORM queries](https://docs.sqlalchemy.org/en/20/orm/session_basics.html#querying), which performs proper validation.